lib/maths/unittest/CTimeSeriesTestForSeasonalityTest.cc

```diff
@@ -977,33 +997,362 @@ BOOST_AUTO_TEST_CASE(testSyntheticWithPiecewiseLinearTrend) {
+        LOG_DEBUG(<< "test " << test + 1 << " / 10");
+
+        std::size_t index{test % generators.size()};
+        auto generator = generators[index];
+        core_t::TTime startTime{
+            HALF_HOUR * (static_cast<core_t::TTime>(startTimes[test]) / HALF_HOUR)};
+
+        values.assign(3 * WEEK / HALF_HOUR, TFloatMeanAccumulator{});
+        for (core_t::TTime time = 0; time < 3 * WEEK; time += HALF_HOUR) {
+            values[time / HALF_HOUR].add(10.0 * generator(startTime + time));
+        }
+
+        maths::CTimeSeriesTestForSeasonality seasonality{startTime, startTime,
+                                                         HALF_HOUR, values};
+        auto result = seasonality.decompose();
+
+        // Expect agreement with generator.
+        predictions.assign(values.size(), 0.0);
+        for (const auto& component : result.seasonal()) {
+            BOOST_REQUIRE_EQUAL(startTime, component.initialValuesStartTime());
+            BOOST_REQUIRE_EQUAL(startTime + 3 * WEEK, component.initialValuesEndTime());
+            for (std::size_t i = 0; i < component.initialValues().size(); ++i) {
+                predictions[i] +=
+                    maths::CBasicStatistics::mean(component.initialValues()[i]);
+            }
+        }
+        for (core_t::TTime time = 0; time < 3 * WEEK; time += HALF_HOUR) {
+            BOOST_REQUIRE_CLOSE_ABSOLUTE(10.0 * generator(startTime + time),
+                                         predictions[time / HALF_HOUR], 1e-4);
+        }
+
+        // Check the seasonal time is initialized correctly.
+        switch (index) {
+        case 0:
+        case 1:
+            for (const auto& component : result.seasonal()) {
+                auto time = component.seasonalTime();
+                BOOST_REQUIRE_EQUAL(false, time->windowed());
+                BOOST_REQUIRE_EQUAL(0, time->windowRepeatStart());
+                BOOST_REQUIRE_EQUAL(0, time->windowStart());
+                BOOST_REQUIRE_EQUAL(WEEK, time->windowEnd());
+                BOOST_REQUIRE_EQUAL(DAY, time->period());
+            }
+            break;
+        default:
+            BOOST_REQUIRE_EQUAL(4, result.seasonal().size());
+            for (std::size_t i = 0; i < result.seasonal().size(); ++i) {
+                auto time = result.seasonal()[i].seasonalTime();
+                BOOST_REQUIRE_EQUAL(true, time->windowed());
+                BOOST_REQUIRE_EQUAL(5 * DAY / HALF_HOUR, time->windowRepeatStart() / HALF_HOUR);
+                BOOST_REQUIRE_EQUAL(expectedWindowStarts[i], time->windowStart());
+                BOOST_REQUIRE_EQUAL(expectedWindowEnds[i], time->windowEnd());
+                BOOST_REQUIRE_EQUAL(expectedPeriods[i], time->period());
+            }
+            break;
+        }
+    }
+}
+
+BOOST_AUTO_TEST_CASE(testNewComponentInitialValuesWithPiecewiseLinearScaling) {
+
+    // Test that the initial values for the seasonal components when there
+    // are linear scalings in the test values.
+}
```

Comment:
Test is missing here?

Reply:
Yes indeed.

Reply:
Done in [c6efe87](https://github.com/elastic/ml-cpp/pull/1451/commits/c6efe87cf3bed3be46d4194cf795f72c4b0f688d).